framework/encode/capture_manager.cpp

```diff
@@ -1048,26 +1051,56 @@ bool CommonCaptureManager::CreateCaptureFile(format::ApiFamilyId api_family, con
     return success;
 }
 
-void CommonCaptureManager::ActivateTrimming()
+void CommonCaptureManager::ActivateTrimming(std::shared_lock<ApiCallMutexT>& current_lock)
 {
-    capture_mode_ |= kModeWrite;
+    auto has_shared_lock = current_lock.owns_lock();
+    if (has_shared_lock)
+    {
+        current_lock.unlock();
```

Comment:
By the time `unlock()` returns and until `AcquireExclusiveApiCallLock()` gets the `unique_lock` there's no locking at all. Can this cause problems?

Reply:
Not in this case. Other threads can acquire the shared lock up until the current thread acquires the unique lock, regardless of whether the current thread does or does not hold the shared lock. The current thread just needs to acquire the unique lock so that it can prevent other threads from modifying API state while it transitions from tracking mode to writing mode and writes the state snapshot, to avoid a potential race condition around the mode transition.